src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1370,7 +1370,8 @@ impl EnumMemberDescriptionFactory<'ll, 'tcx> {
                                 .wrapping_sub(niche_variants.start().as_u32() as u128)
                                 .wrapping_add(niche_start);
                             let value = truncate(value, niche.value.size(cx));
-                            Some(value as u64)
+                            assert_eq!(value as u64 as u128, value);
```

Comment:
Remove `as u64`?

Reply:
Good point, `as u64` seems unnecessary, thanks!

Reply:
Ah, I thought compiler would have optimized it but looking on the history of this line I see it was reasonable for `u64`. Anyway this assert is pointless when `value` has `u128` type.

Reply:
Oh, I got it. I removed `assert`.